apps/api/src/auth/services/api-key/api-key-auth.service.ts

```diff
@@ -0,0 +1,43 @@
+import isAfter from "date-fns/isAfter";
+import parseISO from "date-fns/parseISO";
+import { singleton } from "tsyringe";
+
+import { ApiKeyOutput, ApiKeyRepository } from "@src/auth/repositories/api-key/api-key.repository";
+import { ApiKeyGeneratorService } from "./api-key-generator.service";
+
+@singleton()
+export class ApiKeyAuthService {
+  constructor(
+    private readonly apiKeyGenerator: ApiKeyGeneratorService,
+    private readonly apiKeyRepository: ApiKeyRepository
+  ) {}
+
+  async validateApiKeyFromHeader(apiKey: string | undefined): Promise<ApiKeyOutput | undefined> {
+    if (!apiKey) {
+      return undefined;
+    }
+
+    const [prefix, type, env] = apiKey.split(".");
+    if (prefix !== "ac" || type !== "sk" || env !== (process.env.NODE_ENV === "production" ? "live" : "test")) {
```

Comment:
this will always be `production` in remote env. `DEPLOYMENT_ENV` could be used. Also there're different modules' `ConfigService`s one of which (I think `Core` one) should contain the needed value.